src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -157,4 +174,85 @@ private static Map<String, String> getBackwardsCompatableNames()
 
         return names;
     }
+
+    /**
+     * This class is used to provide backwards compatable support for the settings table in case the {@link Config}
+     * metadata changes. This class will provide the old names for the properties, but will use the new name to
+     * get the value from the {@link Config} object.
+     * <p>
+     * Updating a configuration property object will throw an exception if you will try to update a deprecated property.
+     */
+    private static class BackwardsCompatableRegistry implements Registry
```

Comment:
this code is more verbose than just handling the `Replacement`... before we would create a valid `Property` for each `Replacement` and that would handle the conversions and call the "real" `Property`

It looks like this class makes invalid assumptions, such as old names are not allowed... they are 100% supported, which makes this more verbose it seems...

Is there a reason to migrate away from the existing logic that makes old/new properties look the same?

Reply:
There are a few reasons under the hood to move away from the `ForwardingProperty` logic for the `SettingsTable`:
- The design assumes that we have a central point of access to the `Config' instance - the `ConfigurationRegistry' - to perform mutate configuration operations, ensuring that they are changed in the same way for each public interface;
- We want to restrict the places where the `Config' instance is accessed - the `DatabaseDescriptor' and the `ConfigurationRegistry' are enough to force the `SettingsTable` to do what it needs to do;
- the `Replacement` provides us with the name and type conversions for properties and as there are no usages of the old names in the source code we are able to perform name conversions for the `SettingsTable` needs;

Maybe I'm missing something, but I see the following access path as more verbose than giving short cuts from the `SettingsTable' directly to the `Config':
Config -> DatabaseDescription -> ConfigurationRegistry -> ReplacementRegistry -> SettingsTable;